docs/documentation/head/server-prepare.md

```diff
@@ -9,37 +9,187 @@ nexttitle: Physical and Logical replication API
 `PreparedStatement` interface.
 
-### Note
+> PostgreSQL 9.2 release notes: prepared statements used to be optimized once, without any knowledge
+of the parameters' values. With 9.2, the planner will use specific plans regarding to the parameters
+sent (the query will be planned at execution), except if the query is executed several times and
+the planner decides that the generic plan is not too much more expensive than the specific plans.
+
+Server side prepared statements can improve execution speed as
+1. It sends just statement handle (e.g. `S_1`) instead of full SQL text
+1. It enables use of binary transfer (e.g. binary int4, binary timestamps, etc); the parameters and results are much faster to parse
+1. It enables the reuse server-side execution plan
+1. The client can reuse result set column definition, so it does not have to receive and parse metadata on each execution
+
+### Activation
 
 > Previous versions of the driver used PREPARE and EXECUTE to implement
 server-prepared statements.  This is supported on all server versions beginning
 with 7.3, but produced application-visible changes in query results, such as
 missing ResultSet metadata and row update counts. The current driver uses the V3
-protocol-level equivalents which avoid these changes in query results, but the
-V3 protocol is only available beginning with server version 7.4. Enabling server-prepared
-statements will have no affect when connected to a 7.3 server or when explicitly
-using the V2 protocol to connect to a 7.4 server.
-
-There are a number of ways to enable server side prepared statements depending on
-your application's needs. The general method is to set a threshold for a
-`PreparedStatement`. An internal counter keeps track of how many times the
-statement has been executed and when it reaches the threshold it will start to
-use server side prepared statements.
-
-### Note
-
-> Server side prepared statements are planned only once by the server. This avoids
-the cost of replanning the query every time, but also means that the planner
-cannot take advantage of the particular parameter values used in a particular
-execution of the query. You should be cautious about enabling the use of server
-side prepared statements globally.
+protocol-level equivalents which avoid these changes in query results.
+
+The driver uses server side prepared statements **by default** when `PreparedStatement` API is used.
+In order to get to server-side prepare, you need to execute the query 5 times (that can be
+configured via `prepareThreshold` connection property).
+An internal counter keeps track of how many times the statement has been executed and when it
+reaches the threshold it will start to use server side prepared statements.
+
+Even though reusing of the same `PreparedStatement` object good for performance reasons, the driver
+is able to server-prepare statements automatically across `connection.prepareStatement(...)` calls.
+
```

Comment:
Should be "Even though reuse of the same PreparedStatement object is good"
That being said I'm still unsure what this statement is trying to convey ?

Reply:
The idea is to highlight difference between

```
PreparedStatement st = con.prepare("...");
ps.executeQuery();
..
// later in code
ps.executeQuery();
...

// later in code
ps.executeQuery();
```

vs

```
PreparedStatement st = con.prepare("...");
ps.executeQuery();
ps.close()
..
// later in code
PreparedStatement st = con.prepare("...");
ps.executeQuery();
ps.close()
...

// later in code
PreparedStatement st = con.prepare("...");
ps.executeQuery();
ps.close()
```

Certain JDBC drivers require to use exactly the same `PreparedStatement` object in order to use server-prepared statements.

Reply:
Seems this detailed explanation might be appropriate